core/tests/test_utils.py

```diff
@@ -480,8 +481,12 @@ def mock_search(self, body=None, index=None, params=None):
         terms = body['query']['bool']['must']
 
         for f in filters:
-            for k, v in f['match'].items():
-                result_docs = [doc for doc in result_docs if doc[k] in v]
+            if index == blog_services.SEARCH_INDEX_BLOG_POSTS:
+                for k, v in f['match'].items():
+                    result_docs = [doc for doc in result_docs if v in doc[k]]
+            else:
+                for k, v in f['match'].items():
+                    result_docs = [doc for doc in result_docs if doc[k] in v]
```

Comment:
`doc[k]` can only be of string if `v` is string for  processing `doc[k] in v`. However in case of matching tags for blog posts, `doc[k]` is a list of all the tags assigned to a blog post and `v` is a single tag (type string) which if present in list of tags should return true. Therefore it is `v in doc[k]`. When we perform search for exploration and collections, doc[k] is a single language or category  to which the exploration/collection belong whereas v is a string with languages/categories being separated by  space between them (eg. 'en hi') belonging to which the explorations/collections are required. (Hence `doc[k] in v`)

Reply:
Hmm, okay, please add some explanatory comments in the code.

Reply:
Done.

Reply:
Done